include/verilated_vpi.cpp

```diff
@@ -2355,6 +2344,63 @@ bool vl_check_format(const VerilatedVar* varp, const p_vpi_value valuep, const c
     return status;
 }
 
+bool vl_check_format(const VerilatedVar* varp, const p_vpi_arrayvalue arrayvaluep, const char* fullname, bool isGetValue){
+    //variable must be an array (dims > 1)
+    if (!(varp->dims() >1)){
+        VL_VPI_ERROR_(__FILE__, __LINE__, "%s: Unsupported dimensions (%u)", __func__, varp->dims());
+        return false;
+    }
+
+    if (arrayvaluep->format == vpiIntVal) {
+        switch (varp->vltype()) {
+            case VLVT_UINT8: return true;
+            case VLVT_UINT16: return true;
+            case VLVT_UINT32: return true;
+            default: return false;
```

Comment:
Please make sure tests cover all of these cases -  you can e.g. use nodist/code_coverage to check.